drivers/sdp3x/sdp3x.c

```diff
@@ -0,0 +1,459 @@
+ */
+static int8_t _SDP3x_read_data(const sdp3x_t *dev, int16_t *data)
+{
+    int ret = 0;
+    uint8_t readData[9];
+
+    i2c_acquire(DEV_I2C);
+    ret = i2c_read_bytes(DEV_I2C, DEV_ADDR, readData, sizeof(readData), 0);
+    i2c_release(DEV_I2C);
+
+    if (ret != 0) {
+        DEBUG("[SDP3x] read_data: ret %i\n", ret);
+        return ret;
+    }
+    data[2] = (readData[6] << 8) + readData[7];
+    data[1] = (readData[3] << 8) + readData[4];
+    data[0] = (readData[0] << 8) + readData[1];
+
+    uint8_t correct = 1;
+
+    correct &= _checkCRC(data[2], readData[8]);
+    correct &= _checkCRC(data[1], readData[5]);
+    correct &= _checkCRC(data[0], readData[2]);
+
+    if (correct == 0) {
+        DEBUG("[SDP3x] read_data: CRC-Failure\n");
+        return SDP3x_CRCERROR;
+    }
+
+    return 0;
+}
+
+/**
+ *      intern method to convert sensor value to pascal
+ *      @param  dev      device
+ *      @param  value    raw sensor value
+ *      @return pressure value in centiPa
+ */
+static int32_t _SDP3x_convert_to_pascal(int16_t value, int16_t dp_scale_factor)
+{
+    /* Multiplying by 100 to convert to centiPa */
+    return ((value * 100) / dp_scale_factor);
+}
+
+/**
+ *      intern method to convert sensor value to celsius
+ *      @param  dev     device
+ *      @param  value   raw sensor value
+ *      @return celsius value * 100
+ */
+static int32_t _SDP3x_convert_to_celsius(int16_t value)
+{
+    int16_t div = 200;
+    /* Multiplying by 100 to convert the final temperature in 100*°C */
+    return ((value * 100) / div);
+}
+
+/**
+ *      check if crc is valid
+ *      @param  value Number whose crc has to be calculated
+ *      @param  test  CRC value with which it will compared
+ *      @return 1 on success
+ */
+static int8_t _checkCRC(uint16_t value, uint8_t test)
+{
+    uint8_t crc;
+    uint8_t data[2] = { value >> 8, value & 0x00FF };
+
+    crc = crc8(data, 2, 0x31, 0xFF);
+
+    return (crc == test ? 1 : 0);
+}
+
+/**
+ *      Callback function to handle trigger on irq
+ *      @param  arguments passed when interrupt is raised (in this case sdp3x dev)
+ */
+static void _sdp3x_irq_callback(void *arg)
+{
+    sdp3x_t *dev = (sdp3x_t *)arg;
+    if (!dev->continuous_measurement) {
+        mutex_unlock(&(dev->mutex));
+    }
+}
+
+/**
+ *      Function to check if the product number set
+ *      is the one we get from the sensor
+ *      @param  data  data read from the sensor
+ *      @param  dev   sdp3x device
+ *      @return true  if values from sensor were correct
+ */
+static bool _check_product_number(uint8_t *readData, sdp3x_t *dev)
+{
+    if (readData[0] != 0x03) {
+        return false;
+    }
+    if (readData[1] != 0x01) {
+        return false;
+    }
+    if (dev->params.model == SDP3X_MODEL_31) {
+        if (readData[3] != 0x01) {
+            return false;
+        }
+    } else if (dev->params.model == SDP3X_MODEL_32) {
+        if (readData[3] != 0x02) {
+            return false;
+        }
+    }
```

Comment:
This is the only place where `params.model` is used now.
We don't need to check it just for checking's sake.